File: golabs/godata/go-gumball-riak/gumball-riak/src/github.com/basho/taste-of-riak/java/Ch02-Schemas-and-Indexes/src/SipOfRiak/Item.java
package SipOfRiak;

public class Item
{
    public Item() { }
    public Item(String itemId, String title, Double price) {
        ItemId = itemId;
        Title = title;
        Price = price;
    }
    public String ItemId;
    public String Title;
    public Double Price;
}

File: golabs/godata/go-gumball-riak/gumball-riak/src/github.com/basho/taste-of-riak/csharp/Ch02-Schemas-and-Indexes/Item.cs
namespace Ch02_Schemas_and_Indexes
{
    using System;

    public class Item
    {
        private readonly string id;
        private readonly string title;
        private readonly decimal price;

        public Item(string id, string title, decimal price)
        {
            if (string.IsNullOrWhiteSpace(id))
            {
                throw new ArgumentNullException("id");
            }
            this.id = id;

            if (string.IsNullOrWhiteSpace(title))
            {
                throw new ArgumentNullException("title");
            }
            this.title = title;

            this.price = price;
        }

        public string Id
        {
            get { return id; }
        }

        public string Title
        {
            get { return title; }
        }

        public decimal Price
        {
            get { return price; }
        }
    }
}
